<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Untitled • aphylo</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Untitled">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">aphylo</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.99</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/data_imputation.html">Untitled</a>
    </li>
    <li>
      <a href="../articles/peeling_phylo.html">Peeling Algorithm in Phylogenic Trees</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/USCbiostats/phylogenetic">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Untitled</h1>
                        <h4 class="author">George G. Vega Yon</h4>
            
            <h4 class="date">4/26/2017</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/USCbiostats/phylogenetic/blob/master/vignettes/data_imputation.rmd"><code>vignettes/data_imputation.rmd</code></a></small>
      <div class="hidden name"><code>data_imputation.rmd</code></div>

    </div>

    
    
<div id="definitions" class="section level1">
<h1 class="hasAnchor">
<a href="#definitions" class="anchor"></a>Definitions</h1>
<p>An Annotated Phylogenetic Tree <span class="math inline">\(\tau \equiv (N, E, A)\)</span> is a tuple with nodes <span class="math inline">\(N \equiv \{1 ,\dots, n\}\)</span>, edges <span class="math inline">\(E\equiv \{(i, j) \in N\times N: i\rightarrow j\}\)</span>, where <span class="math inline">\(\rightarrow\)</span> denotes ‘parent of’, and annotations <span class="math inline">\(A \equiv \{a_{ip} \in \{0,1\}: i\in N, p = \{1,\dots,p\}\}\)</span>, where <span class="math inline">\(P\)</span> denotes the number of annotations per node. Furthermore, we say that this tree is a Partially Ordered Annotated Phylogenetic Tree if it’s nodes’ labels form a partial order, this is <span class="math inline">\(i &lt; j, \forall (i, j) \in E\)</span>.</p>
</div>
<div id="predicting-leafs-annotations" class="section level1">
<h1 class="hasAnchor">
<a href="#predicting-leafs-annotations" class="anchor"></a>Predicting Leafs Annotations</h1>
<p>The ultimate goal of this model is to be able to predict, or rather, impute functional states to leaf nodes for which we do not have information. Using both, the information that we have about the annotated tree and the parameter estimates of our model, we can calculate what is the probability of a leaf <span class="math inline">\(i\)</span> having or not a function <span class="math inline">\(p\)</span>. Formally, with <span class="math inline">\(S\equiv (N, E)\)</span>, the probability of leaf node <span class="math inline">\(i\)</span> having the function <span class="math inline">\(p\)</span>, conditional on the tree structure <span class="math inline">\(S^*\)</span> and the observed set of annotations <span class="math inline">\(A^*\)</span>, which excludes annotations on <span class="math inline">\(i\)</span>, is:</p>
<p><span class="math display">\[
\tag{1}\label{eq:1}
{{\mbox{Pr}\left(a_i = 1\;|\;S = S^*, A = A^*\right) }} = \frac{{{\mbox{Pr}\left(A = A^*\;|\;S = S^*, a_i = 1\right) }}\times{{\mbox{Pr}\left(a_i=1\;|\;S = S^*\right) }}}{
{{\mbox{Pr}\left(A = A^*\;|\;S = S^*\right) }}
}
\]</span></p>
<p>For ease of notation, we will drop the <span class="math inline">\(S = S^*\)</span>, as all probabilities throughout this section are conditional on the tree structure. Now, we know that</p>
<p><span class="math display">\[
\tag{2}\label{eq:2}
{{\mbox{Pr}\left(A = A^*\right) }} = {{\mbox{Pr}\left(A = A^*\;|\;a_i = 1\right) }}\times{{\mbox{Pr}\left(a_i = 1\right) }} + {{\mbox{Pr}\left(A = A^*\;|\;a_i = 0\right) }}\times(1 - {{\mbox{Pr}\left(a_i = 1\right) }})
\]</span></p>
<p>Hence, pluging in  into , and multiplying both numerator and denominator by <span class="math inline">\(1/{{\mbox{Pr}\left(a_i = 1\right) }}\)</span>,  can be rewritten as</p>
<p><span class="math display">\[
\tag{1'}
{{\mbox{Pr}\left(a_i = 1\;|\;A = A^*\right) }} = \frac{{{\mbox{Pr}\left(A = A^*\;|\;a_i = 1\right) }}}{
{{\mbox{Pr}\left(A = A^*\;|\;a_i = 1\right) }} + {{\mbox{Pr}\left(A = A^*\;|\;a_i = 0\right) }}\times\frac{\left(1 - {{\mbox{Pr}\left(a_i = 1\right) }}\right)}{{{\mbox{Pr}\left(a_i = 1\right) }}}
}
\]</span> Where <span class="math inline">\({{\mbox{Pr}\left(A = A^*\;|\;a_i = 1\right) }}\)</span> is the likelihood of the data given that we set the ith node to be 1, which we already know how to compute. This way, we are only missing <span class="math inline">\({{\mbox{Pr}\left(a_i=1\right) }}\)</span>, which we can compute as follows:</p>
<p><span class="math display">\[
\begin{aligned}
{{\mbox{Pr}\left(a_i = 1\right) }} &amp; = \pi {{\mbox{Pr}\left(a_i = 1\;|\;a_0 = 1\right) }} + (1 - \pi) {{\mbox{Pr}\left(a_i = 1\;|\;a_0 = 0\right) }}
\end{aligned}
\]</span></p>
<p>Where, given that <span class="math inline">\(\gamma_{01}\)</span> is the shorest path length between node 0 and node <span class="math inline">\(i\)</span>, we have:</p>
<p><span class="math display">\[
\left[\begin{array}{cc}
{{\mbox{Pr}\left(a_i = 0\;|\;a_0 = 0\right) }} &amp; {{\mbox{Pr}\left(a_i = 1\;|\;a_0 = 0\right) }} \\
{{\mbox{Pr}\left(a_i = 0\;|\;a_0 = 1\right) }} &amp; {{\mbox{Pr}\left(a_i = 1\;|\;a_0 = 1\right) }}
\end{array}
\right]
=
\left[\begin{array}{cc}
1 &amp; 0 \\
0 &amp; 1
\end{array} \right] \times
\left[\begin{array}{cc}
1 - \hat\mu_0 &amp;  \hat\mu_0 \\
\hat\mu_1 &amp;  1 - \hat\mu_1
\end{array}
\right]^{\gamma_{0i}}
\]</span></p>
<p>Finally, we will denote the predicted probability as <span class="math inline">\(\hat a_i = {{\mbox{Pr}\left(a_i = 1\;|\;S = S^*, A = A^*\right) }}\)</span></p>
</div>
<div id="assesment-of-model-predictions" class="section level1">
<h1 class="hasAnchor">
<a href="#assesment-of-model-predictions" class="anchor"></a>Assesment of Model Predictions</h1>
<p>Let <span class="math inline">\(A \equiv \{a_{ip}\}_{i\in N, p \in P}\)</span> be an array of size <span class="math inline">\(|N| \times |P|\)</span> are the latent annotations and <span class="math inline">\(Z \equiv \{z_i \in \{0,1,9\}: i\in N\}\)</span> is the experimental data (some missings, 9), we measure the quality of the imputation as</p>
<p><span class="math display">\[
\left\| A_m - \hat A_m \right\|^\mathbf{t} \mathbf{W}^{-1}_m\left\| A_m - \hat A_m \right\|
\]</span></p>
<p>where <span class="math inline">\(A_m \equiv\{a_i \in A: z_i = 9\}\)</span>, the real annotations of the imputed genes, <span class="math inline">\(\tilde A_m \equiv \{\hat a_i: z_i = 9\}\)</span> the predicted annotations of the imputed genes, and <span class="math inline">\(\mathbf{W}\)</span> is a weighting matrix of size <span class="math inline">\(|A_m|^2\)</span>, in particular, the shortest path length between all nodes with <span class="math inline">\(z_i = 9\)</span>.</p>
<p>It can be shown that the previous expression can be written as</p>
<p><span class="math display">\[
\delta = \sum_{i,j}\left[\sum_{p,r} (a_{ip} - \hat a_{ir})^2(a_{jr} - \hat a_{jr})^2\right]^{1/2}w_{ij}^{-1}
\]</span></p>
<p>So, <span class="math inline">\(\mbox{E}\left(\delta\right)\)</span> reduces to computing the following</p>
<p><span class="math display">\[
\sum_{i,j} w_{ij}^{-1}\mbox{E}\left[\sum_{p,r} (a_{ip} - \hat a_{ir})^2(a_{jr} - \hat a_{jr})^2\right]^{1/2}
\]</span></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#definitions">Definitions</a></li>
      <li><a href="#predicting-leafs-annotations">Predicting Leafs Annotations</a></li>
      <li><a href="#assesment-of-model-predictions">Assesment of Model Predictions</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by George Vega Yon.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
